models/comments_models.js

```diff
@@ -20,3 +22,52 @@ exports.fetchCommentByReviewId = (review_id) => {
     })
   })
 }
+
+exports.addCommentByReviewId = (review_id, username, body) => {
+    if (!body) {
+      return Promise.reject({
+        status: 400,
+        msg: "Bad request: Enter a valid comment",
+      });
+    }
+  
+    if (!username) {
+      return Promise.reject({
+        status: 400,
+        msg: "Bad request: Please enter a username",
+      });
+    }
+    if (typeof body !== "string" || typeof username !== "string") {
+      return Promise.reject({
+        status: 400,
+        msg: "Bad request: Incorrect valid data type",
+      });
+    }
+    const queryString = `
+      INSERT INTO comments
+        (review_id, author, body)
+      VALUES
+        ($1, $2, $3)
+      RETURNING *;
+      `;
+  
+    const values = [review_id, username, body];
+  
+    return fetchAllUsers().then((users) => {
```

Comment:
I understand that querying the users first means that you can have a very specific error message, however if you tried to add a row with a non-existent user psql would reject anyway and you could handle that elsewhere, you're kind of creating some unnecessary blocking code here as a result.